app/models/service_order.rb

```diff
@@ -0,0 +1,15 @@
+class ServiceOrder < ActiveRecord::Base
+  belongs_to :tenant
+  belongs_to :user
+  has_many :miq_requests, :dependent => :nullify, :after_add => :assign_user_name_to
+
+  validates :state, :inclusion => {:in => %w(wish cart ordered)}
+  validates :name, :presence => true
+  validates :state, :presence => true
+
+  default_value_for :user_id, User.current_user.id
+
+  def assign_user_name_to(miq_request)
+    miq_request.user_name = user.name
```

Comment:
@gmcculloug does it matter if we use the `User.current_user` for the service order?  Is it possible to create a service order for someone else, such as sharing a wish list with someone?